composer/loggers/mosaicml_logger.py

```diff
@@ -150,6 +150,10 @@ def close(self, state: State, logger: Logger) -> None:
         if self._enabled:
             wait(self._futures)  # Ignore raised errors on close
 
+    def log_checkpoint_upload_start(self, state: State) -> None:
+        self._log_metadata({'checkpoint_upload_start_time': [time.time(), state.timestamp.batch.to_timestring()]})
```

Comment:
I don't really like explicitly hard coding a key to mosaicml_logger. Why not just call log_metadata and then flush at end user?